packages/flame/lib/src/components/sprite_animation_group_component.dart

```diff
@@ -0,0 +1,67 @@
+import 'dart:ui';
+
+import '../extensions/vector2.dart';
+import '../sprite_animation.dart';
+import 'sprite_animation_component.dart';
+
+export '../sprite_animation.dart';
+
+class SpriteAnimationGroupComponent<T> extends SpriteAnimationComponent {
+  /// Key with the current playing animation
+  T? current;
+
+  /// Map with the available states for this animation group
+  late Map<T, SpriteAnimation> animations;
+
+  /// Creates a component with an empty animation which can be set later
+  SpriteAnimationGroupComponent({
+    required this.animations,
+    Vector2? position,
+    Vector2? size,
+    this.current,
+    Paint? overridePaint,
+    bool removeOnFinish = false,
```

Comment:
What does it mean for this to remove when finishing? Can it ever finish, since it can change states?

Reply:
Ah, it means that once a "child" finishes it is removed, but if the state is chosen again it will start it again?

Reply:
Yeah, whenever a state that don't loop finishes and this Flag is true, the component will be removed.

At first I thought this would be a very edge case, but I think this is quite useful. Imagine a spaceship component and one of its state is it exploding. With this Flag true and the explosion state not being a loop would make life very easy.

Reply:
I don't think I quite get it still. The whole group here is `removeOnFinish`, isn't the explosion just one of the states within the group?

Reply:
Yeah, I mean the way it is implemented right now, this flag is kind of a global flag for all states, any state that doesn't loop will make the component be removed.

Do you think we should have a flag per state?

Reply:
Not sure what is more clear, we should explain it in the docs at least.
Your are thinking that the most used implementation will be that the initial state is a looping one and that any non-looping state should remove the component if it swaps to that one?
I think to me it would be more clear to have the flag per state.

Reply:
Yes that was what I was thinking.

But to be sure and cover all the cases I will create a flag per state then.

To do that I think it will not be worth that this class extends SpriteAnimatiomComponent anymore, as they will differ too much, but I don't think this is a problem.

Tomorrow I will do that follow up.

Reply:
Cool cool!